haystack/components/embedders/types/protocol.py

```diff
@@ -0,0 +1,30 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, Protocol, TypeVar
+
+T = TypeVar("T", bound="TextEmbedder")
+
+# See https://github.com/pylint-dev/pylint/issues/9319.
+# pylint: disable=unnecessary-ellipsis
+
+
+class TextEmbedder(Protocol):
+    """
+    Protocol for Text Embedders.
+    """
+
+    def run(self, text: str) -> Dict[str, Any]:
```

Comment:
Just a comment: As "embedding" as a key is mandatory and any other keys in the dict are optional, I thought about making the protocol more explicit. We could try the following:

```python
class EmbeddingOutput(TypedDict, total=False):
    """
    The output structure for text embedders.
    The 'embedding' field is required, all other fields are optional.
    """
    embedding: List[float]
    # Allow any additional string keys with any value type
    __extra__: NotRequired[Dict[str, Any]]
```

and then 
`def run(self, text: str) -> EmbeddingOutput:` 

However, all the embedders we currently have type their output only as ` -> Dict[str, Any]` or don't define it at all. We shouldn't have a protocol that is more strict than the current implementations.

Reply:
So, let's keep this as is.

Reply:
> currently have type their output only as -> Dict[str, Any]

Exactly, that was the main reason why I left it like that